plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -139,4 +193,266 @@ import NotificationCenter
+    }
+
+    @objc(pushRegister:)
+    func pushRegister(command: CDVInvokedUrlCommand) {
+        Optimove.shared.pushRequestDeviceToken()
+
+        self.commandDelegate.send(CDVPluginResult(status: .ok), callbackId: command.callbackId)
+    }
+
+    @objc(inAppUpdateConsent:)
+    func inAppUpdateConsent(command: CDVInvokedUrlCommand) {
+        let consented: Bool = command.arguments[0] as! Bool
+        OptimoveInApp.updateConsent(forUser: consented)
+
+        self.commandDelegate.send(CDVPluginResult(status: .ok), callbackId: command.callbackId)
+    }
+
+    @objc(inAppGetInboxItems:)
+    func inAppGetInboxItems(command: CDVInvokedUrlCommand) {
+        self.commandDelegate.run(inBackground:{
+            let inboxItems = OptimoveInApp.getInboxItems()
+            var items = [[String : Any]]()
+
+            let formatter = DateFormatter()
+            formatter.timeStyle = .full
+            formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+            formatter.timeZone = TimeZone(secondsFromGMT: 0)
+
+            for item in inboxItems {
+                var dict: [String: Any] = [
+                    "id": item.id,
+                    "title": item.title,
+                    "subtitle": item.subtitle,
+                    "availableFrom": item.availableFrom != nil ? formatter.string(from: item.availableFrom!) : "",
+                    "availableTo": item.availableTo != nil ? formatter.string(from: item.availableTo!) : "",
+                    "dismissedAt": item.dismissedAt != nil ? formatter.string(from: item.dismissedAt!) : "",
+                    "isRead": item.isRead(),
+                    "sentAt": formatter.string(from: item.sentAt)
+                ]
+
+                if let data = item.data {
+                    dict["data"] = data
+                }
+
+                if let imageUrl = item.getImageUrl() {
+                    dict["imageUrl"] = imageUrl.absoluteString
+                }
+
+                items.append(dict)
+            }
+
+
+            let pluginResult = CDVPluginResult(status: .ok, messageAs: items)
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+        })
+    }
+
+    @objc(inAppPresentInboxMessage:)
+    func inAppPresentInboxMessage(command: CDVInvokedUrlCommand) {
+        self.commandDelegate.run(inBackground: {
+            let messageId = command.arguments[0] as! Int64
+            let inboxItems = OptimoveInApp.getInboxItems()
+
+            var pluginResult = CDVPluginResult.init(status: CDVCommandStatus_ERROR, messageAs: "Message not found or not available")
+            for msg in inboxItems {
+                if (msg.id != messageId){
+                    continue
+                }
+
+                let result = OptimoveInApp.presentInboxMessage(item: msg)
+                if (result == .PRESENTED){
+                    pluginResult = CDVPluginResult.init(status: CDVCommandStatus_OK)
+                }
```

Comment:
Is there any else to reject? Not sure what Android side of plugin did in this case.

Reply:
Ah nvm, see handled by default above (Y)